File: imagegame_ui/src/services/PlayerStatisticsService.js
import http from "../http-common"; 

class PlayerStatisticsService {
  getAllPlayerStatisticss(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/playerStatistics/playerStatisticss`, searchDTO);
  }

  get(playerStatisticsId) {
    return this.getRequest(`/playerStatistics/${playerStatisticsId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/playerStatistics?field=${matchData}`, null);
  }

  addPlayerStatistics(data) {
    return http.post("/playerStatistics/addPlayerStatistics", data);
  }

  update(data) {
  	return http.post("/playerStatistics/updatePlayerStatistics", data);
  }
  
  uploadImage(data,playerStatisticsId) {
  	return http.postForm("/playerStatistics/uploadImage/"+playerStatisticsId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PlayerStatisticsService();

File: src/main/java/com/imagegame/service/PlayerStatisticsService.java
package com.imagegame.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.imagegame.domain.PlayerStatistics;
import com.imagegame.dto.PlayerStatisticsDTO;
import com.imagegame.dto.PlayerStatisticsSearchDTO;
import com.imagegame.dto.PlayerStatisticsPageDTO;
import com.imagegame.dto.PlayerStatisticsConvertCriteriaDTO;
import com.imagegame.service.GenericService;
import com.imagegame.dto.common.RequestDTO;
import com.imagegame.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PlayerStatisticsService extends GenericService<PlayerStatistics, Integer> {

	List<PlayerStatistics> findAll();

	ResultDTO addPlayerStatistics(PlayerStatisticsDTO playerStatisticsDTO, RequestDTO requestDTO);

	ResultDTO updatePlayerStatistics(PlayerStatisticsDTO playerStatisticsDTO, RequestDTO requestDTO);

    Page<PlayerStatistics> getAllPlayerStatisticss(Pageable pageable);

    Page<PlayerStatistics> getAllPlayerStatisticss(Specification<PlayerStatistics> spec, Pageable pageable);

	ResponseEntity<PlayerStatisticsPageDTO> getPlayerStatisticss(PlayerStatisticsSearchDTO playerStatisticsSearchDTO);
	
	List<PlayerStatisticsDTO> convertPlayerStatisticssToPlayerStatisticsDTOs(List<PlayerStatistics> playerStatisticss, PlayerStatisticsConvertCriteriaDTO convertCriteria);

	PlayerStatisticsDTO getPlayerStatisticsDTOById(Integer playerStatisticsId);







}





